src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ExecuteTaskReconfigurationServiceIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+    @SpyBean
+    CFTTaskDatabaseService cftTaskDatabaseService;
+    @Autowired
+    TransactionTemplate transactionTemplate;
+
+    @SpyBean
+    CaseConfigurationProviderService caseConfigurationProviderService;
+
+    List<TaskResource> taskResources;
+
+    @BeforeEach
+    void setUp() {
+        executeTaskReconfigurationService = new ExecuteTaskReconfigurationService(
+            cftTaskDatabaseService,
+            taskReconfigurationHelper
+        );
+
+        taskResources = taskResourceRepository.findAllByTaskIdIn(
+            List.of("8d6cc5cf-c973-11eb-bdba-0242ac222009",
+                  "8d6cc5cf-c973-11eb-bdba-0242ac222019",
+                  "8d6cc5cf-c973-11eb-bdba-0242ac222029"), Sort.by(Sort.Order.asc("taskId"))
+        );
+    }
+
+    @Test
+    void should_retry_changes_when_exception_during_reconfigure_task_resource() {
+        TaskResource taskResource1 = taskResources.get(0);
+        assert taskResource1 != null;
+        taskResource1.setTitle("title1");
+        doReturn(taskResource1).when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource1);
+
+        TaskResource taskResource2 = taskResources.get(1);
+        assert taskResource2 != null;
+        taskResource2.setTitle("title1");
+        doThrow(new OptimisticLockException("locked"))
+            .when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource2);
+
+        TaskResource taskResource3 = taskResources.get(2);
+        assert taskResource3 != null;
+        taskResource3.setTitle("title1");
+        doReturn(taskResource3).when(taskAutoAssignmentService).reAutoAssignCFTTask(taskResource3);
+
+        TaskConfigurationResults results = new TaskConfigurationResults(
+            emptyMap(),
+            configurationDmnResponse(),
+            permissionsResponse()
+        );
+        doReturn(results).when(caseConfigurationProviderService)
+            .getCaseRelatedConfiguration(anyString(), anyMap(), anyBoolean());
+        List<TaskFilter<?>> taskFilters = createReconfigureTaskFilters();
+        TaskOperationRequest request = new TaskOperationRequest(
+            TaskOperation.builder()
+                .type(TaskOperationType.EXECUTE_RECONFIGURE)
+                .retryWindowHours(1L)
+                .maxTimeLimit(30)
+                .runId("")
+                .build(),
+            taskFilters
+        );
+
+        try {
+            executeTaskReconfigurationService.performOperation(request);
+        } catch (TaskExecuteReconfigurationException e) {
+            log.error("TaskExecuteReconfigurationException Exception {}", e.getMessage());
+        }
+
+        //doInTransaction(() -> executeTaskReconfigurationService.performOperation(request));
+        verify(cftTaskDatabaseService, times(2)).saveTask(any(TaskResource.class));
+        // 6 times because the taskResource2 will be retried
+        verify(taskReconfigurationHelper, times(6)).reconfigureTaskResource(anyString());
```

Comment:
Think the verifications can be more specific here using the Task id?
How many times should we retry each taskId in this scenario